riscv/sim.cc

```diff
@@ -74,6 +70,20 @@ sim_t::sim_t(const cfg_t *cfg, bool halted,
 
   debug_module.add_device(&bus);
 
+  socketif = NULL;
+#ifdef HAVE_BOOST_ASIO
+  if (socket_enabled) {
+    socketif = new socketif_t();
+  }
+#else
+  if (socket_enabled) {
+    fputs("Socket support requires compilation with boost asio; "
+          "please rebuild the riscv-isa-sim project without --without-boost-asio\n",
```

Comment:
This could be confusing to a user who doesn't have Boost installed, and did not use that option to `configure`.

Would it be correct to say `rebuild with --with-boost-asio`? Then `configure` will tell them that the Boost package is missing. (Right?)